conftest.py

```diff
@@ -0,0 +1,11 @@
+import pytest
+from django.contrib.auth import get_user_model
+
+CustomUser = get_user_model()
+
+pytestmark = pytest.mark.django_db
+
+@pytest.fixture
+def book_owner1():
+    user1 = CustomUser.objects.filter(pk=1) 
+    return user1 if user1.exists() else CustomUser.objects.create_user(username="user1",password="letmein",id=1,is_superuser=True)
```

Comment:
There are couple things to take away here:

1. As with the general comments, avoid hard-coding ids in your fixtures. It's ok for a fixture to just create a user every time it runs. Since each test runs in a transaction that gets rolled back, your user will be destroyed every time anyway.

2. Avoid if/else on the same line if it's a large statement. Also be careful about whether an ORM method returns an object or a queryset. You could rewrite the same statement as:
```python
user = CustomUser.objects.filter(pk=1).first() # notice the first() here too, otherwise filter() returns a queryset, not a single instance!
if not user:
    user = CustomUser.objects.create_user(
        username="user1", password="letmein", id=1, is_superuser=True)
return user
```

A way to re-factor this would be:

```python
def book_owner(django_user_model):
    return django_user_model.objects.create(username="book_owner", password="letmein", is_superuser=True)
```

However, also consider this:
What makes the fixture user a "book owner"? Nothing really, it's just a user without any assigned books or anything else. Could this be named "admin_user" instead? Is there a reason to have this fixture at all? Maybe it could be replaced by pytest-django's `admin_user` or `admin_client` (so that you can skip manually logging in too).

Reply:
I had it like  the first version first, but thought it would be more concise with ternary, but agree, it's a bit long! The second one looks great. 

Ultimately I should probably get rid of the whole default stuff. But it was good to do it to learn about the conftest.py and that you can write a fixture you can use everywhere... also about how foreign key relationships work in the code.